### ISSUE TITLE
DecodeError: Error parsing message

### ISSUE THREAD

**Issue by jorgeecr** [soCzech/TransNetV2#1]
Hello, I'm trying to run the model but I get the following error 
```
020-06-22 00:20:45.880699: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1
[TransNetV2] Using weights from transnetv2-weights/.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/saved_model/loader_impl.py", line 98, in parse_saved_model
    saved_model.ParseFromString(file_content)
google.protobuf.message.DecodeError: Error parsing message

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "transnetv2.py", line 188, in <module>
    main()
  File "transnetv2.py", line 160, in main
    model = TransNetV2(args.weights)
  File "transnetv2.py", line 17, in __init__
    self._model = tf.saved_model.load(model_dir)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/saved_model/load.py", line 578, in load
    return load_internal(export_dir, tags)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/saved_model/load.py", line 588, in load_internal
    loader_impl.parse_saved_model_with_debug_info(export_dir))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/saved_model/loader_impl.py", line 56, in parse_saved_model_with_debug_info
    saved_model = _parse_saved_model(export_dir)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/saved_model/loader_impl.py", line 101, in parse_saved_model
    raise IOError("Cannot parse file %s: %s." % (path_to_pb, str(e)))
OSError: Cannot parse file b'transnetv2-weights/saved_model.pb': Error parsing message.
```

Any idea on what could be wrong? thanks

**Comment by soCzech** [comment]
Does file `transnetv2-weights/saved_model.pb` exist? It's on `git-lfs` so maybe `git clone` did not download it.

**Comment by soCzech** [comment]
Ok, I retested the code by running the following and it works perfectly:
```
git --version
# >> git version 2.20.1 (older versions do not download lfs files automatically)
git clone git@github.com:soCzech/TransNetV2.git
cd TransNetV2/
# nvidia-docker is required!
docker build -t transnet -f inference/Dockerfile .
docker run -it --rm --gpus 1 -v /home/videos:/tmp transnet transnetv2_predict /tmp/test_video.mp4 --visualize
```
You may need to redownload the `transnetv2-weights` directory as the files could be corrupted.

Here are sha256 sums for check (get them by running `sha256sum _file_`):
```
8ac2a52c5719690d512805b6eaf5ce12097c1d8860b3d9de245dcbbc3100f554  saved_model.pb
b8c9dc3eb807583e6215cabee9ca61737b3eb1bceff68418b43bf71459669367  variables/variables.data-00000-of-00001
8b99e28b4ad11372d9a1ad9703298c2e370df14859da4245fdbe818e92dd403f  variables/variables.index
```

Did that solve your issue?

**Comment by jorgeecr** [comment]
Thanks a lot for the quick answer. I was git cloning and I did get different files (got different sha256 sums). After downloading the files manually I got the correct sums and everything worked perfectly. Thanks again for the help and for the great model!

**Comment by Petryashin** [comment]
Hello! I have the following problem:
When I initialize the model  "model = TransNetV2()" An error appears related to the weights:
OSError: Cannot parse file b'C:\\ProgramData\\Anaconda3\\envs\\deeplearning\\lib\\site-packages\\transnetv2-1.0.0-py3.8.egg\\transnetv2\\transnetv2-weights/saved_model.pb': Wrong wire type in tag..

The above exception was the direct cause of the following exception:

DecodeError                               Traceback (most recent call last)
C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\tensorflow\python\saved_model\loader_impl.py in parse_saved_model(export_dir)
     97       file_content = file_io.FileIO(path_to_pb, "rb").read()
---> 98       saved_model.ParseFromString(file_content)
     99       return saved_model

C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\google\protobuf\message.py in ParseFromString(self, serialized)
    198     self.Clear()
--> 199     return self.MergeFromString(serialized)
    200 

C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\google\protobuf\internal\python_message.py in MergeFromString(self, serialized)
   1144     try:
-> 1145       if self._InternalParse(serialized, 0, length) != length:
   1146         # The only reason _InternalParse would return early is if it

C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\google\protobuf\internal\python_message.py in InternalParse(self, buffer, pos, end)
   1197         old_pos = new_pos
-> 1198         (data, new_pos) = decoder._DecodeUnknownField(
   1199             buffer, new_pos, wire_type)  # pylint: disable=protected-access

C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\google\protobuf\internal\decoder.py in _DecodeUnknownField(buffer, pos, wire_type)
    992   else:
--> 993     raise _DecodeError('Wrong wire type in tag.')
    994 

DecodeError: Wrong wire type in tag.

During handling of the above exception, another exception occurred:

OSError                                   Traceback (most recent call last)
C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\transnetv2-1.0.0-py3.8.egg\transnetv2\transnetv2.py in __init__(self, model_dir)
     17         try:
---> 18             self._model = tf.saved_model.load(model_dir)
     19         except OSError as exc:

C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\tensorflow\python\saved_model\load.py in load(export_dir, tags, options)
    602   """
--> 603   return load_internal(export_dir, tags, options)
    604 

C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\tensorflow\python\saved_model\load.py in load_internal(export_dir, tags, options, loader_cls)
    613   saved_model_proto, debug_info = (
--> 614       loader_impl.parse_saved_model_with_debug_info(export_dir))
    615 

C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\tensorflow\python\saved_model\loader_impl.py in parse_saved_model_with_debug_info(export_dir)
     55   """
---> 56   saved_model = _parse_saved_model(export_dir)
     57 

C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\tensorflow\python\saved_model\loader_impl.py in parse_saved_model(export_dir)
    100     except message.DecodeError as e:
--> 101       raise IOError("Cannot parse file %s: %s." % (path_to_pb, str(e)))
    102   elif file_io.file_exists(path_to_pbtxt):

OSError: Cannot parse file b'C:\\ProgramData\\Anaconda3\\envs\\deeplearning\\lib\\site-packages\\transnetv2-1.0.0-py3.8.egg\\transnetv2\\transnetv2-weights/saved_model.pb': Wrong wire type in tag..

The above exception was the direct cause of the following exception:

OSError                                   Traceback (most recent call last)
<ipython-input-65-5b86b11b3e4c> in <module>
----> 1 model = TransNetV2()#model_dir = 'C:\\ProgramData\Anaconda3\\envs\\deeplearning\\Lib\\site-packages\\transnetv2-1.0.0-py3.8.egg\\transnetv2\\transnetv2-weights/')

C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\transnetv2-1.0.0-py3.8.egg\transnetv2\transnetv2.py in __init__(self, model_dir)
     18             self._model = tf.saved_model.load(model_dir)
     19         except OSError as exc:
---> 20             raise IOError(f"[TransNetV2] It seems that files in {model_dir} are corrupted or missing. "
     21                           f"Re-download them manually and retry. For more info, see: "
     22                           f"https://github.com/soCzech/TransNetV2/issues/1#issuecomment-647357796") from exc

OSError: [TransNetV2] It seems that files in C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\transnetv2-1.0.0-py3.8.egg\transnetv2\transnetv2-weights/ are corrupted or missing. Re-download them manually and retry.

Please tell me how to solve this problem.

**Comment by soCzech** [comment]
Read the error message or read the comments above ^^^

> OSError: [TransNetV2] It seems that files in C:\ProgramData\Anaconda3\envs\deeplearning\lib\site-packages\transnetv2-1.0.0-py3.8.egg\transnetv2\transnetv2-weights/ are corrupted or missing. **Re-download them manually and retry.**

**Comment by NXTNiklas** [comment]
Hello, I've got the same Issue and was able to solve it with the following commands:

git clone https://github.com/soCzech/TransNetV2.git
cd TransNetV2
git lfs fetch https://github.com/soCzech/TransNetV2.git
git lfs checkout

The problem is, that the files are to big for normal storing and are stored in githubs large file storage (lfs). When running on colab like I did, it only downloads the pointer files. The commands above solve that.

**Comment by Scr-w** [comment]
Hello, is there a link where I can manually download the weight, because this weight always times out when my server pulls it. I look forward to hearing from you soon. Thank you very much.

**Comment by mk12306** [comment]
Hello, where can i find the  'sha256sum _file_'? Thank you very much.

**Comment by soCzech** [comment]
https://github.com/soCzech/TransNetV2/tree/master/inference/transnetv2-weights

**Comment by youngazier** [comment]
Hi, I have used [NXTNiklas](https://github.com/NXTNiklas)'s solution (because I don't know how to use docker) but got the same error. 
OSError: [TransNetV2] It seems that files in "(my root path)../transnetv2/transnetv2-weights/" are corrupted or missing. Re-download them manually and retry. For more info, see: https://github.com/soCzech/TransNetV2/issues/1#issuecomment-647357796
I also checked sha256 sums numbers from [soCzech](https://github.com/soCzech) and they are the same. Can anyone help me please? Thanks.

> Hello, I've got the same Issue and was able to solve it with the following commands:
> 
> git clone https://github.com/soCzech/TransNetV2.git cd TransNetV2 git lfs fetch https://github.com/soCzech/TransNetV2.git git lfs checkout
> 
> The problem is, that the files are to big for normal storing and are stored in githubs large file storage (lfs). When running on colab like I did, it only downloads the pointer files. The commands above solve that.

**Comment by youngazier** [comment]
I have fixed it by downgrade python to 3.7 and tensorflow to 2.1 :))) 

> Hi, I have used [NXTNiklas](https://github.com/NXTNiklas)'s solution (because I don't know how to use docker) but got the same error. OSError: [TransNetV2] It seems that files in "(my root path)../transnetv2/transnetv2-weights/" are corrupted or missing. Re-download them manually and retry. For more info, see: [#1 (comment)](https://github.com/soCzech/TransNetV2/issues/1#issuecomment-647357796) I also checked sha256 sums numbers from [soCzech](https://github.com/soCzech) and they are the same. Can anyone help me please? Thanks.
> 
> > Hello, I've got the same Issue and was able to solve it with the following commands:
> > git clone https://github.com/soCzech/TransNetV2.git cd TransNetV2 git lfs fetch https://github.com/soCzech/TransNetV2.git git lfs checkout
> > The problem is, that the files are to big for normal storing and are stored in githubs large file storage (lfs). When running on colab like I did, it only downloads the pointer files. The commands above solve that.

**Comment by Prasannad02** [comment]
@soCzech  INFO:     127.0.0.1:54986 - "GET / HTTP/1.1" 200 OK
e corrupted or missing. Re-download them manually and retry. For more info, see: https://github.com/soCzech/TransNetV2/issues/1#issuecomment-647357796      
ERROR:main:ValueError: Scene detection failed: [TransNetV2] It seems that files in TransNetV2/transnetv2-weights are corrupted or missing. Re-download them manually and retry. For more info, see: https://github.com/soCzech/TransNetV2/issues/1#issuecomment-647357796
INFO:     127.0.0.1:54997 - "POST /upload-video/ HTTP/1.1" 400 Bad Request how to solve this
